File: Frontend/Mayaar.Web/src/types/category.ts
export interface Category {
    id: string;
    name: string;
    description?: string | null;
    imageUrl?: string | null;
    isActive: boolean;
    createdAt: string;
    updatedAt?: string | null;
}

export interface CreateCategoryRequest {
    name: string;
    description?: string | null;
    imageUrl?: string | null;
}

export interface UpdateCategoryRequest {
    name: string;
    description?: string | null;
    imageUrl?: string | null;
    isActive: boolean;
}

File: Backend/Mayaar/Mayaar.Domain/Entities/Category.cs
namespace Mayaar.Domain.Entities;

/// <summary>
/// Represents a product category in Mayaar.
/// </summary>
public sealed class Category
{
    /// <summary>
    /// Gets or sets the unique identifier.
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// Gets or sets the category name.
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the category description.
    /// </summary>
    public string? Description { get; set; }

    /// <summary>
    /// Gets or sets the URL of the category image.
    /// </summary>
    public string? ImageUrl { get; set; }

    /// <summary>
    /// Gets or sets a value indicating whether the category is active.
    /// </summary>
    public bool IsActive { get; set; } = true;

    /// <summary>
    /// Gets or sets the creation date and time in UTC.
    /// </summary>
    public DateTime CreatedAt { get; set; }

    /// <summary>
    /// Gets or sets the last modification date and time in UTC.
    /// </summary>
    public DateTime? UpdatedAt { get; set; }
}